### PR TITLE
Update manifests README to discuss self-hosted.

### PR DESCRIPTION
**Pull request by caseydavenport**
This adds some documentation for the self-hosted work.

Relies on a new release of the CNI plugin which includes https://github.com/projectcalico/calico-cni/pull/177

### CODE REVIEW

**Comment by caseydavenport** [comment]
@tomdee Do you think this is the right place for the configuration documentation, or should that live in the calico-cni repo?

File: docs/cni/kubernetes/manifests/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Calico Kubernetes Hosted Install
+
+This directory contains Kubernetes manifests to deploy Calico on top of Kubernetes.
+
+- `calico-configmap.yaml`: Contains a Kubernetes ConfigMap for configuring the deployment.  Make sure the values
+in this file match your desired configuration.
+
+- `calico-hosted.yaml`: Contains a Kubernetes DaemonSet which install and runs Calico on each Kubernetes master and node.
```
Comment (tomdee) [inline docs/cni/kubernetes/manifests/README.md]: installs

File: docs/cni/kubernetes/manifests/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# How it works
+
+The `calico-hosted.yaml` file contains all the necessary resources for installing Calico on each node in your Kubernetes cluster.
+
+It does the following things:
+- Installs the `calico/node` container on each host using a DaemonSet.
+- Installs the Calico CNI binaries and network config on each host using a DaemonSet.
+- Runs the `calico/kube-policy-controller` pod as a ReplicaSet.
+
+# Configuration options
```
Comment (tomdee) [inline docs/cni/kubernetes/manifests/README.md]: Should this be ##

Comment (caseydavenport) [inline docs/cni/kubernetes/manifests/README.md]: I don't think so

**Comment by caseydavenport** [comment]
Circle failure looked like a dockerhub issue failing to pull calico/test.

**Pull request merged**